platform/darwin/src/MGLVectorTileSource.mm

```diff
@@ -96,7 +96,7 @@ + (NSString *)preferredMapboxStreetsLanguage {
             mostSpecificLanguage = language;
         }
     }
-    return mostSpecificLanguage ?: @"en";
+    return mostSpecificLanguage;
```

Comment:
This fallback turned out to be redundant: `preferredLanguages` is never empty, so `mostSpecificLanguage` is never `nil`.

`+[NSBundle preferredLocalizationsFromArray:forPreferences:]` never returns an empty array of languages, because it always falls back to English if none of the available locales matches one of the preferred locales. To maximize the flexibility of the label localization functionality, I’ve been thinking it would be better if we fall back to `name` rather than `name_en`. That way, for instance, if a user prefers Hebrew – which is supported by the map SDK’s UI – they can at least get Hebrew in Israel rather than English everywhere.

The trick will be to distinguish between the case where `+[NSBundle preferredLocalizationsFromArray:forPreferences:]` returns English because the locale really does match English (`en`, `en-US`, `eng-AU`, etc.) from the case where it chose English as a last resort. I’m open to doing this as a followup task, potentially in a patch release, unless a solution comes to mind.

/cc @friedbunny @nickidlugash @langsmith